README.md

```diff
@@ -6,99 +6,155 @@
+- `module`*
+
+\\* Any module that implements the `Parameter.Field` behaviour is elegible to be a field in the schema definition.
+
+The options available for the field definition are:
+- `key`: This is the key on the external source that will be converted to the param definition.
+As an example, if you receive data from an external source that uses snake case for mapping `first_name`, this flag should be `key: "firstName"`.
+If this parameter is not set it will default to the field name.
+- `default`: default value of the field.
+- `required`: defines if the field needs to be present when parsing the input.
+
+After the definition the schema can be validate and parsed against external parameters using the `Parameter.load/3` function.
+
+## Data Deserialization
+
+This is a common requirement when you receive data from an external source and wants to
+validate and deserialize this data to an Elixir definition. This can be achived using `Parameter.load/2` or `Parameter.load/3` functions:
+
+```elixir
+iex> params = %{
+      "address" => %{"city" => "Tallinn"},
+      "age" => "32",
+      "firstName" => "Paulo",
+      "lastName" => "Curado"
     }
-  ```
+...> Parameter.load(UserSchema, params)
+  %{
+    address: %{city: "Tallinn"},
+    age: 32,
+    first_name: "Paulo",
+    last_name: "Curado"
+  }
+```
+
+Adding invalid data should return validation errors:
+
+```elixir
+iex> params = %{
+      "address" => %{"city" => "Tallinn", "number" => "123AB"},
+      "age" => "AA",
+      "firstName" => "Paulo",
+      "lastName" => "Curado"
+    }
+...> Parameter.load(UserSchema, params)
+{:error, %{address: %{number: "invalid integer type"}, age: "invalid integer type"}}
+```
 
-  Adding invalid data should return validation errors:
+## Custom Types
 
-  ```elixir
-  iex> params = %{
-        "address" => %{"city" => "Tallinn", "number" => "123AB"},
-        "age" => "AA",
-        "firstName" => "Paulo",
-        "lastName" => "Curado"
-      }
-  ...> Parameter.load(UserSchema, params)
-  {:error, %{address: %{number: "invalid integer type"}, age: "invalid integer type"}}
-  ```
+For implementing custom types create a module that implements the  `Parameter.Parametrizable` behaviour.
+
+Check the following example on how `Integer` parameter was implemented:
+
+```Elixir
+defmodule IntegerCustomType do
+  @moduledoc """
+  Integer parameter type
+  """
+
+  @behaviour Parameter.Parametrizable
+
+  @impl true
+  # `load/1` is evaluatign when parsing the parameters, you can do validations here and transform the data
+  def load(value) when is_integer(value) do
+    {:ok, value}
+  end
+
+  def load(value) when is_binary(value) do
+    case Integer.parse(value) do
+      {integer, ""} -> {:ok, integer}
+      _error -> error_tuple()
+    end
+  end
+
+  def load(_value) do
+    error_tuple()
+  end
+
+  @impl true
+  # `validate/1` is checked during compile time and verifies if default value is passed on the schema an validating using this function
```

Comment:
# `validate/1` checks the schema during compile time. it verifies the default value if it's passed to the schema validating its type